internal/edge.go

```diff
@@ -32,55 +31,79 @@ type edgeConfiguration struct {
 	PolicyID string `json:"policyId"`
 }
 
-var edgeConfigurationChan = make(chan *edgeConfiguration, 1)
+// EdgeConnector listens for Edge Thing configuration changes and notifies the corresponding EdgeClient
+type EdgeConnector struct {
+	mqttClient MQTT.Client
+	cfg        *edgeConfiguration
+	edgeClient edgeClient
+}
+
+// edgeClient receives notifications of Edge Thing configuration changes from EdgeConnector
+type edgeClient interface {
+	Connect(client MQTT.Client, scriptSUPConfig *ScriptBasedSoftwareUpdatableConfig, cfg *edgeConfiguration) error
+	Disconnect(closeStorage bool)
+}
 
-// retrieveEdgeConfiguration tries to connect to the local MQTT broker and retrieve its configuration.
-func retrieveEdgeConfiguration(server string, username string, password string) (*edgeConfiguration, MQTT.Client, error) {
-	logger.Infof("Retrieve edge configuration from: %s", server)
+// newEdgeConnector create EdgeConnector with the given server, username and password for the given EdgeClient
```

Comment:
- create -> creates
- with the given server, username and password -> with the given configuration